rclcpp/src/rclcpp/service.cpp

```diff
@@ -39,11 +37,11 @@ ServiceBase::~ServiceBase()
 std::string
 ServiceBase::get_service_name()
 {
-  return this->service_name_;
+  return rcl_service_get_service_name(service_handle_);
```

Comment:
I understand what you're doing here, but it would be better I think to leave it as is and fix all cases of this (subscription, client, etc.) at the same time. This change is not required to fulfill the main purpose of this pr I think.